### PR TITLE
Surface LSIF supported langs

### PR DESCRIPTION
**Pull request by macraig**
Make LSIF indexers we recommend easily accessible from the Code intel doc root.

### CODE REVIEW

**Review by olafurpg — APPROVED** [review]
Good idea!

File: doc/code_intelligence/index.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Code intelligence provides advanced code navigation features that let developers
 
 <div class="cta-group">
 <a class="btn btn-primary" href="explanations/introduction_to_code_intelligence">★ Introduction to code intelligence</a>
+<a class="btn" href="references/indexers.md">LSIF supported languages</a>
```
Comment (emchap) [inline doc/code_intelligence/index.md:11]: Do we want to include the * ahead of the text as on the previous line for stylistic consistency?

**Review by emchap — CHANGES_REQUESTED** [review]

File: doc/code_intelligence/index.md (hunk 1 above)
Comment (emchap) [reply doc/code_intelligence/index.md:11]: (If that's intentional as a difference, ignore me—thank you for doing this!)

Comment (macraig) [reply doc/code_intelligence/index.md:11]: It's intentional as we usually mark the Quickstart or "Intro to..." with the start to indicate the first section a user should look at. Does that make sense?

Comment (emchap) [reply doc/code_intelligence/index.md:11]: Sounds good!

**Review by emchap — APPROVED** [review]

File: doc/code_intelligence/index.md (hunk 1 above)
Comment (macraig) [inline doc/code_intelligence/index.md:11]:
```suggestion
<a class="btn" href="references/indexers">LSIF supported languages</a>
```

**Pull request merged**